docs/content/index.md

```diff
@@ -2,60 +2,153 @@
+the rollup granularity is the minimum granularity you will be able to query data at. Hence, Druid ingestion specs define this granularity as the `queryGranularity` of the data.
+
+Druid is designed to perform single table operations and does not currently support joins.
+Although many production setups instrument joins at the ETL level, data must be denormalized before it is loaded into Druid.
+
+## Indexing the Data
+
+Druid gets its speed in part from how it stores data. Borrowing ideas from search infrastructure,
+Druid creates immutable snapshots of data, stored in data structures highly optimized for analytic queries.
+
+Druid is a column store, which means each individual column is stored separately. Only the columns that pertain to a query are used
+in that query, and Druid is pretty good about only scanning exactly what it needs for a query.
+Different columns can also employ different compression methods. Different columns can also have different indexes associated with them.
+
+## Sharding the Data
+
+Druid shards are called `segments` and Druid always first shards data by time. In our compacted data set, we can create two segments, one for each hour of data.
+
+For example:
+
+Segment `sampleData_2011-01-01T01:00:00:00Z_2011-01-01T02:00:00:00Z_v1_0` contains
+
+     2011-01-01T01:00:00Z  ultratrimfast.com  google.com  Male   USA     1800        25     15.70
+     2011-01-01T01:00:00Z  bieberfever.com    google.com  Male   USA     2912        42     29.18
+
+
+Segment `sampleData_2011-01-01T02:00:00:00Z_2011-01-01T03:00:00:00Z_v1_0` contains
+
+     2011-01-01T02:00:00Z  ultratrimfast.com  google.com  Male   UK      1953        17     17.31
+     2011-01-01T02:00:00Z  bieberfever.com    google.com  Male   UK      3194        170    34.01
+
+Segments are self-contained containers for the time interval of data they hold. Segments
+contain data stored in compressed column orientations, along with the indexes for those columns. Druid queries only understand how to
+scan segments.
+
+Segments are uniquely identified by a datasource, interval, version, and partition number.
+Multiple segments may exist for the same interval of time for the same datasource. These segments form a `block` for an interval.
+Druid queries will only complete if a `block` is complete. That is to say, if a block consists of 3 segments, such as:
+
+`sampleData_2011-01-01T02:00:00:00Z_2011-01-01T03:00:00:00Z_v1_0`
+
+`sampleData_2011-01-01T02:00:00:00Z_2011-01-01T03:00:00:00Z_v1_1`
+
+`sampleData_2011-01-01T02:00:00:00Z_2011-01-01T03:00:00:00Z_v1_2`
+
+All 3 segments must be loaded before a query for the interval `2011-01-01T02:00:00:00Z_2011-01-01T03:00:00:00Z` completes.
+
+## Loading the Data
+
+Druid currently has a couple of different ways of getting data into it. In the near future, we'd like to address this complexity and unify the various data ingestion methods.
```

Comment:
There are a ton of apologies in this section, and I don't think the intro is the place for that. Instead, we could say that Druid has two modes of ingestion (batch and real-time) and then briefly describe the guarantees of each.

I think the one apology we can keep is when we mention that real-time ingestion in Druid is best-effort. At that point we can say that we're planning to address this in the future. I don't think we need to mention other open projects.